internal/featuredetection/feature_detection.go

```diff
@@ -94,25 +99,63 @@ func (d *detector) PullRequestFeatures() (PullRequestFeatures, error) {
 	// }
 
 	features := PullRequestFeatures{}
+	gql := api.NewClientFromHTTP(d.httpClient)
 
-	var featureDetection struct {
-		PullRequest struct {
-			Fields []struct {
-				Name string
-			} `graphql:"fields(includeDeprecated: true)"`
-		} `graphql:"PullRequest: __type(name: \"PullRequest\")"`
-	}
+	g := new(errgroup.Group)
+	g.Go(func() error {
+		var pullRequestFeatureDetection struct {
+			PullRequest struct {
+				Fields []struct {
+					Name string
+				} `graphql:"fields(includeDeprecated: true)"`
+			} `graphql:"PullRequest: __type(name: \"PullRequest\")"`
+		}
 
-	gql := api.NewClientFromHTTP(d.httpClient)
-	err := gql.Query(d.host, "PullRequest_fields", &featureDetection, nil)
-	if err != nil {
-		return features, err
-	}
+		if err := gql.Query(d.host, "PullRequest_fields", &pullRequestFeatureDetection, nil); err != nil {
+			return err
+		}
 
-	for _, field := range featureDetection.PullRequest.Fields {
-		if field.Name == "isInMergeQueue" {
-			features.MergeQueue = true
+		for _, field := range pullRequestFeatureDetection.PullRequest.Fields {
+			if field.Name == "isInMergeQueue" {
+				features.MergeQueue = true
+			}
 		}
+
+		return nil
+	})
+
+	g.Go(func() error {
+		if !ghinstance.IsEnterprise(d.host) {
```

Comment:
This is why I chose to do all the declaration and parsing in separate go routines. If we're just modifying the final feature struct (concurrent safe as long as we don't reference the same field) then we can early exit here.

Reply:
GraphhQL actually allows us to do multiple queries in a single request which I think is a better option here than creating two parallel requests. We used to do something similar until it became unnecessary, but we can bring it back. Here is an example:

https://github.com/cli/cli/pull/5494/files#diff-21986411c01246bfb45a509ecd73874c24989888023920096d6bf7481e533b04L286

Reply:
Ah, I didn't know that thanks. I blindly pulled this pattern from https://github.com/cli/cli/pull/5494/commits/4eec5c2a3de7bcba687d3e165670de57f40bc588#diff-531901056679f44f90e67a1294e67f916dd55000a95452c6716461e9f3286bb9L100-L107